### PR TITLE
SM-323 | feat: update offer event

### PR DESCRIPTION
**Pull request by ilamini**
- [x] Update Offer For ERC20 Tokens
- [x] Update Offer For ERC1155 Tokens
- [x] Cancel Nft Offer

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "generate:config": "echo $NETWORK && mustache config/$NETWORK.json subgraph.yaml.mustache > subgraph.yaml",
     "codegen": "graph codegen --output-dir src/types/",
     "build": "graph build",
-    "deploy": "graph deploy --access-token $ACCESS_TOKEN $GITHUB_USERNAME/$SUBGRAPH_NAME --ipfs https://api.thegraph.com/ipfs/ --node https://api.thegraph.com/deploy/",
+    "deploy": "graph deploy --access-token %ACCESS_TOKEN% %GITHUB_USERNAME%/%SUBGRAPH_NAME% --ipfs https://api.thegraph.com/ipfs/ --node https://api.thegraph.com/deploy/",
```
Comment (ElvisKrop) [inline package.json:8]: @ilamini Please, undo those changes

Comment (ilamini) [reply package.json]: Done

File: src/mappings/dOTCManager.ts [hunk 2]
```diff
@@ -149,3 +152,29 @@ export function handleNftOfferCompleted(event: CompletedNftOffer): void {
...
+}
+
+export function handleTokenOfferUpdated(event: TokenOfferUpdated): void {
+  let offer = Offer.load(event.params.offerId.toHex())
+  if (offer != null) {
+    let tokenOut = Token.safeLoad(offer.tokenOut)
+    offer.amountOut = bigIntToDecimal(event.params.newOffer, tokenOut.decimals)
+    offer.save()
+  }
+}
```
Comment (ElvisKrop) [inline src/mappings/dOTCManager.ts:171]: @ilamini Should the offer.price change also?

Comment (ilamini) [reply src/mappings/dOTCManager.ts]: Correct I have update this as well.

File: src/mappings/dOTCManager.ts [hunk 3]
```diff
@@ -149,3 +152,29 @@ export function handleNftOfferCompleted(event: CompletedNftOffer): void {
...
+
+export function handleCanceledNftOffer(event: CanceledNftOffer): void {
+  let offer = NftOffer.load(event.params.offerId.toHex())
+  if (offer != null) {
+    offer.cancelled = true
+    offer.save()
+  }
+}
+
+export function handleTokenOfferUpdated(event: TokenOfferUpdated): void {
```
[the remark this answers is not in the document]
Comment (ilamini) [reply src/mappings/dOTCManager.ts:164]: So this part is not affected.

**Review by ElvisKrop — APPROVED** [review]

**Pull request merged**